### DISCUSSION TITLE
Alarm Base Station Volume Randomly Changing

### DISCUSSION THREAD

**Discussion by bship810** [tsightler/ring-mqtt#788] [Support Q&A]
Periodically, I am seeing different configurations in my Ring alarm system changing. Most noticeable is the base alarm station volume changing. We keep it at 50% and it will sometimes be at full volume. Its become so frequent that I have created an automation the resets it to the proper volume every 5 minutes whether it needs it or not. I have looked into the logs and can see the incorrect volumes occasionally coming in for the base station. I have tried restarting Ring-MQTT and have tried re-authenticating without any change. Is there a known fix for this? 

Worth noting, I have been using this plugin for a long time now and it was working great, so I'm not sure what is triggering this behavior.

I am using version 5.6.3 of Ring-MQTT and Mosquito Broker 6.4.0.

**Comment by tsightler** [comment]
You state that you can see these incorrect values coming in from the logs, that means that **_something_** has to be sending the command to change the volume, because that function will never be called without a command being received.  Based on this, you will have to determine what is sending the command to change the volume.  My gut would say maybe some automation that turns a room on/off is incorrectly turning the volume to 100 or 0 but, as I have no idea of your environment, only you can troubleshoot it.

Feel free to post the log snippet, it should only be two lines, something like:

```
2024-02-21T17:59:55.643Z ring-mqtt [Alarm Base Station] Received set volume level to 70%
2024-02-21T17:59:55.834Z ring-mqtt [Alarm Base Station] ring/a9ce7234-fafb-421c-4508-29d54f48cbd3/alarm/eb7d8190-3cef-4199-bd3a-91d60a8a9110/volume/state 70
```

**Comment by bship810** [comment]
Thanks for your response. I have uninstalled and reinstalled the add on, and it seems to be stable since doing that.

When looking at the ring-mqtt log, I wouldn't see the "Received set volume level" entry, but I have seen the second line with the volume level being read in with the random number values. I included a screenshot from Home Assistant from a few days ago showing the trend of what it was doing. It was random in what value the volume level was being set. I'm fairly confident it wasn't being set anywhere else from Ring or HA.

![2024-02-22 10_31_53-History – Home Assistant — Mozilla Firefox](https://github.com/tsightler/ring-mqtt/assets/115731792/48a9130d-d7d2-41b0-8ac4-372b137a462e)

**Reply by tsightler** [reply]
I'm not sure what is happening, but the code here is very simple, volume is set by calling a function, setVolumeLevel(), which in turn calls setVolume() and sends the JSON command over the websocket to change the volume.  This path includes logging so any time this is called, it will be logged.

If you are not seeing the "Received set volume level", then the volume setting cannot be coming from ring-mqtt because that code path always logs this message prior to calling the setVolume function, thus the volume changes have to be occurring from something else, and ring-mqtt is simply reporting the changed state.

**Comment by georgedyjr** [comment]
I also experienced this phantom volume changing and couldn't figure out what was causing it. Each time I reset it back to 50%, it would come back to 100%. I didn't see the "Received set volume level..." in the logs and can't imagine what else would be causing it since I have a pretty bare bones HA setup and just started using MQTT yesterday.

@bship810 did the reinstall hold?

<img width="1421" height="554" alt="Screen Shot 2026-03-18 at 9 58 14 PM" src="https://github.com/user-attachments/assets/f2b62076-16bc-4998-a754-2497ca75e4b2" />

**Reply by bship810** [reply]
Hey @georgedyjr,

Yeah, after reinstalling I no longer had the issue.

**Comment by georgedyjr** [comment]
I think I found the problem...maybe? When ring-mqtt starts up, it calls detectVolumeAccess() on my Base Station, which internally calls setVolume() with a test value to probe whether volume control is supported. That test value is coming back as something outside the 0–1 range. The unhandled promise rejection from that crash causes the process to die, the HA watchdog restarts it, and the cycle repeats.

```
WARNING - Unhandled Promise Rejection
Volume must be between 0 and 1
Error: Volume must be between 0 and 1
  at RingDevice.setVolume (ring-device.js:46)
  at BaseStation.detectVolumeAccess (base-station.js:16)
  at new BaseStation (base-station.js:10)
```

**Reply by tsightler** [reply]
I don't understand how the original value can be more than 1 though, which is what is required for this to be the case.  I mean, yes, there is a bug that if somehow the value was, for example, .995, then maybe, but that shouldn't be possible, especially when the value is 50%, which should just be .50, so nowhere close.  Maybe you can add some debigging to see what the original value being reported actually is.

Maybe Ring has changed something with their base stations and how volume is reporting.  What model base station do you have?  Is it a Gen2 maybe?  I don't see how this wouldn't be happening for 1000's of people otherwise and the code is unchanged for ages.
